crates/egui/src/containers/scroll_area.rs

```diff
@@ -439,21 +439,13 @@ impl ScrollArea {
         {
             // Clip the content, but only when we really need to:
             let clip_rect_margin = ui.visuals().clip_rect_margin;
-            let scroll_bar_inner_margin = ui.spacing().scroll_bar_inner_margin;
             let mut content_clip_rect = ui.clip_rect();
             for d in 0..2 {
                 if has_bar[d] {
                     if state.content_is_too_large[d] {
                         content_clip_rect.min[d] = inner_rect.min[d] - clip_rect_margin;
                         content_clip_rect.max[d] = inner_rect.max[d] + clip_rect_margin;
                     }
-
-                    if state.show_scroll[d] {
-                        // Make sure content doesn't cover scroll bars
-                        let tiny_gap = 1.0;
-                        content_clip_rect.max[1 - d] =
-                            inner_rect.max[1 - d] + scroll_bar_inner_margin - tiny_gap;
-                    }
```

Comment:
The `1-d` is not incorrect. When we have a horizontal scroll-bar (`d = 0`), the scrollbar itself is at he bottom of the vertical dimension (1-d), and the other way around.

I believe this code was added to handle nested scroll-areas. Have you tried that?

Reply:
I have not. I'll tinker with it and see what I can find.

Reply:
> I believe this code was added to handle nested scroll-areas. Have you tried that?

Do you have some code to specifically test this?

Reply:
What I mean is something that specifically shows what the change was trying to correct.